sklearn/utils/extmath.py

```diff
@@ -707,6 +707,97 @@ def _safe_accumulator_op(op, x, *args, **kwargs):
     return result
 
 
+def _incremental_weighted_mean_and_var(X, sample_weight,
+                                       last_weighted_mean,
+                                       last_weighted_variance,
+                                       last_weight_sum):
+    """Calculate weighted mean and variance batch update
+
+        last_weighted_mean and last_weighted_variance are statistics
+        computed at the last step by the function. Both must be
+        initialized to 0.0. In case no scaling is required
+        last_weighted_variance can be None. The weighted_mean is
+        always required and returned because necessary for the
+        calculation of the weighted_variance. last_weight sum is
+        the sum of weights encountered until now.
+
+        Derived from the paper "Incremental calculation of
+        weighted mean and variance",
+        by Tony Finch.
+
+        Parameters
+        ----------
+        X : array-like, shape (n_samples, n_features)
+            Data to use for statistics update
+
+        sample_weight : array-like, shape (n_samples,)
+
+        last_weighted_mean : array-like, shape: (n_features,)
```

Comment:
Let's call these `last_mean` and `last_variance` (and similarly `updated_mean` etc). There is no need to indicate in each variables that it's weighted.